inc/class-wpseo-utils.php

```diff
@@ -355,18 +359,26 @@ public static function emulate_filter_bool( $value ) {
 
 		if ( is_bool( $value ) ) {
 			return $value;
-		} else if ( is_int( $value ) && ( $value === 0 || $value === 1 ) ) {
-			return (bool) $value;
-		} else if ( ( is_float( $value ) && ! is_nan( $value ) ) && ( $value === (float) 0 || $value === (float) 1 ) ) {
-			return (bool) $value;
-		} else if ( is_string( $value ) ) {
-			$value = trim( $value );
-			if ( in_array( $value, $true, true ) ) {
-				return true;
-			} else if ( in_array( $value, $false, true ) ) {
-				return false;
+		} else {
+			if ( is_int( $value ) && ( $value === 0 || $value === 1 ) ) {
+				return (bool) $value;
 			} else {
```

Comment:
Oops... thought you'd taken this change out already... Please change it back.
Always stick to the golden rule '_don't nest what doesn't need nesting_'.